test/images/image-util.sh

```diff
@@ -117,23 +141,46 @@ push() {
   docker_version_check
   TAG=$(<"${IMAGE}"/VERSION)
   if [[ -f ${IMAGE}/BASEIMAGE ]]; then
-    archs=$(listArchs)
+    base_images=$(listBaseImages)
+    # NOTE(claudiub): if the REMOTE_DOCKER_URL var is not set, or it is an empty string, we must skip
+    # pushing the Windows image and including it into the manifest list.
+    if [[ ! -v "REMOTE_DOCKER_URL" || -z "${REMOTE_DOCKER_URL}" ]]; then
+      if [[ ! -z "$(printf "%s\n" $base_images | grep '^windows')" ]]; then
+        echo "Skipping pushing the image '${IMAGE}' for Windows. REMOTE_DOCKER_URL should be set, containing the URL to a Windows docker daemon."
+        base_images=$(printf "%s\n" $base_images | grep -v "^windows")
+      fi
+    fi
   else
-    archs=${!QEMUARCHS[@]}
+    # prepend linux/ to the QEMUARCHS items.
+    base_images=`printf 'linux/%s\n' "${!QEMUARCHS[@]}"`
   fi
-  for arch in ${archs}; do
-    docker push "${REGISTRY}/${IMAGE}-${arch}:${TAG}"
+  for base_image in ${base_images}; do
+    if [[ $base_image =~ .*/.* ]]; then
+      os_name=`echo $base_image | cut -d "/" -f 1`
+      arch=`echo $base_image | cut -d "/" -f 2`
+    fi
```

Comment:
Can you handle the missing `else` condition here?

Reply:
There's no ``else`` case. There are 2 cases when building an image: 

- it has  ``BASEIMAGE`` file, Which this PR updates them so **all** of them are os/arch formatted. Also, the ``getBaseImage`` now requires both OS and arch. The grep will fail without both of them now.
- ``BASEIMAGE`` does not exist, in which case ``linux/`` is prepended above on L51.

Reply:
I think it's always a good idea to handle `else` statements for every `if`, no matter what the surrounding state is *supposed* to be (at the very least, you can at least print an error message and exit with an error).

Reply:
Ok, added ``else``.